### PR TITLE
docs: expand logging documentation

### PR DESCRIPTION
**Pull request by smira**
Add notes on pod logs and kernel log delivery.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4622)

### CODE REVIEW

**Comment by smira** [comment]
Preview: https://deploy-preview-4622--talos-dev.netlify.app/docs/v0.14/guides/logging/

File: website/content/docs/v0.14/Guides/logging.md [hunk 1]
```diff
@@ -35,8 +35,28 @@ $ talosctl -n 172.20.1.2 logs machined
...
+NODE         NAMESPACE   ID                                                 IMAGE                                                         PID    STATUS
+172.20.1.2   k8s.io      kube-system/kube-flannel-dk6d5                     k8s.gcr.io/pause:3.5                                          1329   SANDBOX_READY
+172.20.1.2   k8s.io      └─ kube-system/kube-flannel-dk6d5:install-cni      ghcr.io/talos-systems/install-cni:v0.7.0-alpha.0-1-g2bb2efc   0      CONTAINER_EXITED
+172.20.1.2   k8s.io      └─ kube-system/kube-flannel-dk6d5:install-config   quay.io/coreos/flannel:v0.13.0                                0      CONTAINER_EXITED
+172.20.1.2   k8s.io      └─ kube-system/kube-flannel-dk6d5:kube-flannel     quay.io/coreos/flannel:v0.13.0                                1610   CONTAINER_RUNNING
+172.20.1.2   k8s.io      kube-system/kube-proxy-gfkqj                       k8s.gcr.io/pause:3.5                                          1311   SANDBOX_READY
+172.20.1.2   k8s.io      └─ kube-system/kube-proxy-gfkqj:kube-proxy         k8s.gcr.io/kube-proxy:v1.23.0-rc.0                            1379   CONTAINER_RUNNING
+
+$ talosctl -n 172.20.1.2 logs -k
```
Comment (andrewrynhard) [inline website/content/docs/v0.14/Guides/logging.md:50]: missing container name (`talosctl -n 172.20.1.2 logs -k <container>`)? Or do we get all logs in this case?

**Review by andrewrynhard — APPROVED** [review]

File: website/content/docs/v0.14/Guides/logging.md (hunk 1 above)
Comment (smira) [reply website/content/docs/v0.14/Guides/logging.md:50]: nope, it's copy-paste error :facepalm:

Comment (smira) [reply website/content/docs/v0.14/Guides/logging.md]: fixed

**Review by AlekSi — APPROVED** [review]

**Review by frezbo — APPROVED** [review]

**Pull request merged**